packages/neuron-wallet/src/services/wallets.ts

```diff
@@ -221,6 +215,59 @@ export class FileKeystoreWallet extends Wallet {
   }
 }
 
+export class HardwareFromSeedWallet extends FileKeystoreWallet {
+  static fromJSON = (json: WalletProperties) => {
+    return new HardwareFromSeedWallet(json)
+  }
+
+  public toJSON = () => {
+    return {
+      id: this.id,
+      name: this.name,
+      extendedKey: this.extendedKey,
+      device: this.device,
+      isHD: this.isHD,
+      startBlockNumber: this.startBlockNumber,
+      hardwareFromSeed: true,
+    }
+  }
+
+  public checkAndGenerateAddresses = async (): Promise<AddressInterface[] | undefined> => {
+    const { publicKey } = hd.AccountExtendedPublicKey.parse(this.extendedKey)
+    const address = await AddressService.generateAndSaveForPublicKeyQueue.asyncPush({
+      walletId: this.id,
+      publicKey,
+      addressType: hd.AddressType.Receiving,
+      addressIndex: ROOT_ADDRESS_INDEX,
+    })
+
+    if (address) {
+      return [address]
+    }
+  }
+
+  public getNextAddress = async (): Promise<AddressInterface | undefined> => {
+    return AddressService.getFirstAddressByWalletId(this.id)
+  }
+
+  public getNextChangeAddress = async (): Promise<AddressInterface | undefined> => {
+    return AddressService.getFirstAddressByWalletId(this.id)
+  }
+
+  public getNextReceivingAddresses = async (): Promise<AddressInterface[]> => {
+    const address = await AddressService.getFirstAddressByWalletId(this.id)
+    if (address) {
+      return [address]
+    }
+
+    return []
+  }
```

Comment:
These methods should throw because the wallet of the root path won't derive more.

Reply:
These methods are copied from HardwareWallet, and they will be called.